src/transformState.js

```diff
@@ -4,8 +4,33 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const objects of actions) {
+    const { type, extraData, keysToRemove } = objects;
+
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const deleteValue of keysToRemove) {
+          delete state[deleteValue];
+        }
+        break;
+
+      case 'clear' :
+        for (const deleteAll in state) {
```

Comment:
delete all would be a good name for a function, but not for a variable.
It has to be a noun and it should describe what is stored inside